### PR TITLE
fix: fixing dropdown position in flex layouts

### PR DESCRIPTION
**Pull request by bclark-p44**
Closes # 

## 📝 Description

> Add a brief description

## Screenshots

> Please provide screenshots for any visual changes

## Merge checklist

- [ ] Added/updated tests
- [ ] Added changeset

### CODE REVIEW

File: packages/react/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -66,12 +68,31 @@ const Button = React.forwardRef<ButtonElement, ButtonProps>((props, forwardedRef
 
   const buttonRef = React.useRef<HTMLButtonElement>(null);
 
+  const handleClick = React.useCallback(
```
Comment (bclark-p44) [inline packages/react/src/components/Button/Button.tsx:71]: Fixing the click/press handlers to support both click and press events.

File: packages/react/src/components/Combobox/Combobox.tsx [hunk 2]
```diff
@@ -94,16 +95,6 @@ const Combobox = React.forwardRef<ComboboxElement, ComboboxProps>((props, forwar
     within: true,
   });
 
-  const { overlayProps: positionProps } = useOverlayPosition({
```
Comment (bclark-p44) [inline packages/react/src/components/Combobox/Combobox.tsx:97]: moved this downstream into the `Popover`

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -28,65 +28,104 @@ interface DropdownProps {
...
   return (
     <DropdownContext.Provider
       value={{
         closeOnSelect,
         menuProps: mergeProps(menuProps, { autoFocus: state.focusStrategy || true }),
         menuRef,
+        onClose: state.close,
       }}
     >
-      <PressResponder {...menuTriggerProps} ref={menuTriggerRef} isPressed={state.isOpen}>
```
Comment (bclark-p44) [inline packages/react/src/components/Dropdown/Dropdown.tsx:86]: Removing the `PressResponder` in favor of just using the child directly.

File: packages/react/src/components/Popover/Popover.tsx [hunk 4]
```diff
@@ -53,57 +101,87 @@ interface PopoverProps extends DOMProps, StyleProps {
...
+        onExited: handleExited,
+        overlayProps,
+        overlayRef,
+        placement,
+        positionProps,
+        shouldCloseOnBlur,
+        shouldCloseOnInteractOutside,
+      }}
+    >
+      {mounted && (
```
Comment (bclark-p44) [inline packages/react/src/components/Popover/Popover.tsx:175]: OK here is where the fix comes in. I needed to ensure the hooks in `PopoverContent` gets called after the `OverlayContainer` to preserve the call cycle of the hooks.

File: packages/react/src/components/internal/DropdownItem/DropdownItem.tsx [hunk 5]
```diff
@@ -65,13 +67,18 @@ const DropdownItem: React.FC<DropdownItemProps> = props => {
...
     {
       'aria-label': item['aria-label'],
+      closeOnSelect,
       key,
       isDisabled,
-      isVirtualized,
       isSelected,
+      isVirtualized,
+      onAction,
+      onClose,
```
Comment (bclark-p44) [inline packages/react/src/components/internal/DropdownItem/DropdownItem.tsx:81]: forgot to ensure that clicking an item in the dropdown closes the dropdown.

File: packages/react/package.json [hunk 6]
```diff
@@ -27,6 +27,7 @@
   "dependencies": {
     "@internationalized/date": "^3.0.0",
     "@project44-manifest/design-tokens": "^1.1.0",
+    "@radix-ui/react-slot": "^1.0.0",
```
Comment (Ashley-p44) [inline packages/react/package.json:30]: ohhh we are still using radix?

File: packages/react/src/components/Provider/Provider.context.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import * as React from 'react';
+
+export type ColorMode = 'light' | 'dark';
+
+export interface ProviderContext {
```
Comment (Ashley-p44) [inline packages/react/src/components/Provider/Provider.context.ts:5]: is this a generic provider?

File: packages/react/src/components/Select/Select.tsx [hunk 8]
```diff
@@ -139,7 +130,7 @@ const Select = React.forwardRef<SelectElement, SelectProps>((props, forwardedRef
     >
       <div className={classes} ref={forwardedRef}>
         {startIcon && (
-          <span className={cx('manifest-select--icon', 'manifest-select--icon--start')}>
+          <span className={cx('manifest-select__icon', 'manifest-select__icon--start')}>
```
Comment (Ashley-p44) [inline packages/react/src/components/Select/Select.tsx:133]: why the change in classname? Is this due to the update?

File: packages/react/src/components/internal/PopoverContent/PopoverContent.tsx [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+              <DismissButton onDismiss={onClose} />
+            </div>
+          )}
+        </Transition>
+      </FocusScope>
+    );
+  },
+);
+
+if (__DEV__) {
```
Comment (Ashley-p44) [inline packages/react/src/components/internal/PopoverContent/PopoverContent.tsx:102]: nice

**Review by Ashley-p44 — APPROVED** [review]
Just had a couple questions, but looks good.

File: packages/react/package.json (hunk 6 above)
Comment (bclark-p44) [reply packages/react/package.json:30]: Only slots, since its a good tool for cloning and passing props to children.

File: packages/react/src/components/Provider/Provider.context.ts (hunk 7 above)
Comment (bclark-p44) [reply packages/react/src/components/Provider/Provider.context.ts:5]: oh this is future work. let me delete it

File: packages/react/src/components/Select/Select.tsx (hunk 8 above)
Comment (bclark-p44) [reply packages/react/src/components/Select/Select.tsx:133]: yeah it was a break from a previous commit
